pythran/pythonic/random/shuffle.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef PYTHONIC_RANDOM_SHUFFLE_HPP
+#define PYTHONIC_RANDOM_SHUFFLE_HPP
+
+#include "pythonic/include/random/shuffle.hpp"
+
+#include "pythonic/utils/proxy.hpp"
+#include "pythonic/random/random.hpp"
+
+#include <limits>
+
+namespace pythonic {
+
+    namespace random {
+
+        template <class T>
+        void shuffle(T & seq) {
+            std::shuffle(seq.begin(), seq.end(), __random_generator);
+        }
+
+        template <class T, class function>
+        void shuffle(T & seq, function && randf) {
+            struct URG {
```

Comment:
If you define the URG here, you will create a template version for each T type (so list<long> and list<float> will create 2 differentes URG version while they only depend on function).
Also, this is an implementation details so you should put it in a local "details" namespace and replace its call with something like
`details::URG<function>(std::forward<function>(randf))` (You may looks for details namespace in others part of the code).